cdap-ui/app/features/admin/controllers/NamespaceDatasetMetadataController.js

```diff
@@ -1,15 +1,27 @@
 angular.module(PKG.name + '.feature.admin').controller('AdminNamespaceDatasetMetadataController',
-function ($scope, $state, $alert, MyDataSource) {
+function ($scope, $state, $alert, MyDataSource, $filter) {
 
   var dataSrc = new MyDataSource($scope);
 
   dataSrc.request({
     _cdapPath: '/namespaces/' + $state.params.nsadmin
-                  + '/data/explore/tables/dataset_' + $state.params.datasetId + '/info'
-  }).then(function (res) {
-    $scope.metadata = res;
+                  + '/data/explore/tables'
+  }).then(function (tables) {
+    var match = $filter('filter')(tables, $state.params.datasetId);
+    if (match.length > 0) {
+      dataSrc.request({
+        _cdapPath: '/namespaces/' + $state.params.nsadmin
```

Comment:
It's weird that we call this variable `nsadmin`. It's a string that is the namespace, right?